pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherSingleActiveConsumer.java

```diff
@@ -393,6 +397,16 @@ protected void readMoreEntries(Consumer consumer) {
             // active-cursor reads message from cache rather from bookkeeper (2) if topic has reached message-rate
             // threshold: then schedule the read after MESSAGE_RATE_BACKOFF_MS
             if (serviceConfig.isDispatchThrottlingOnNonBacklogConsumerEnabled() || !cursor.isActive()) {
+                //If it reaches the threshold of the entire Broker rate limit, try to read again later.
+                if (topic.getBrokerService().getBrokerDispatchRateLimiter() != null
+                        && topic.getBrokerService().getBrokerDispatchRateLimiter().isDispatchRateLimitingEnabled()
+                        && topic.getBrokerService().getBrokerDispatchRateLimiter().isConsumeRateExceeded()) {
```

Comment:
I have achived this function in my local source in a similar way but use a precise rate-limiter like 'DispatchRateLimiter'.
And I found a problem that consume rate limiter fail to limit the consume rate for partitioned topic when limit the bytes rate.
It seems this PR also have same problem, did you have a test for partitioned topic with bytes rate limit in a cluster? (for example consume a 50 partitioned topic)

The problem is: 
when multi consumer start to consume at same time, 'isConsumeRateExceeded()' here return false for all consumers. And at this timing we do not know the byte size of message we will consume, we have to call 'incrementConsumeCount()' after all these messages deliverd. That will cause a large traffic of consume when a partitioned topic consumer start.
And even 'incrementConsumeCount()' was called after all these messages deliverd, the consume count will reset to 0 every second. In the next second 'isConsumeRateExceeded()' will still return false, that will make the real consume rate much larger the configed rate.

Reply:
All current limiting in Pulsar will have this problem:
We judge whether the threshold is exceeded before reading the entry; but increasing the count of msg and byte is after reading data, so there will be a time difference.
Even if we use the precise Limiter, the count will not increase until the data is read. If you read concurrently in an instant, the threshold will still be exceeded.
MultiConsumer will create multiple Consumers to consume each partition at the same time, which is considered as a concurrent read.

If we want to be absolutely accurate, we may need to make Cursor judge whether it exceeds the threshold every time it reads a message. I think this is a huge loss in performance.

Reply:
Yes, I also found current topic level `DispatchRateLimiter` take no effect when multi subscriptions are created for this topic to do concurrent read. But for broker level consume,  concurrent read happens all the time, which will make broker level rate limiter take no effect.
I have changed the behaviour of `RateLimter`, increase the byte count after data is read, and minus threshold byte every second instead of reset count to 0. Thus the reading will stop for a while and the final broker level threshold could be limited as expect.
I did not test the loss of performance in this way, and still can not solve the problem which cause large traffic of consume at the moment of starting the consumer, which is mentioned in #7720